registry/warming.go

```diff
@@ -169,6 +226,36 @@ func (w *Warmer) warm(id image.Name, creds Credentials) {
 	}
 	awaitFetchers.Wait()
 	w.Logger.Log("updated", id.String())
+
+	if w.Notify != nil {
+		// If there's more tags than there used to be, there must be
+		// at least one new tag.
+		if len(cacheTags) < len(tags) {
+			w.Notify()
+			return
+		}
+		// Otherwise, check whether there are any entries in the
+		// fetched tags that aren't in the cached tags, ignoring any
+		// in the cached tags that aren't in fetched tags.
+		sort.Strings(cacheTags)
```

Comment:
It's not super obvious from just glancing at the code why exactly this section works. I figured it out because I have context but maybe it could do with comments in the switch statement and the last if statement.

Reply:
That is a fair point! I can't say I agonised over this but you are right, it does deserve explanation.